src/Voron/Data/BTrees/Tree.MultiTree.cs

```diff
@@ -30,8 +31,191 @@ namespace Voron.Data.BTrees
      */
     public unsafe partial class Tree
     {
+        private static void ValidateValuesForMultiBulkAdd(int maxNodeSize, ReadOnlySpan<Slice> values)
```

Comment:
Is this method required to be executed on every call, like it is possible that we could end up in a situation in which a call to this aint a catastrophic operation? 
What I am asking is: It would be crazy to just let this one run on DEBUG/VALIDATE instead to catch issues and avoid having to do it every time on Release?